ocaml-lsp-server/test/e2e/__tests__/textDocument-codeAction.test.ts

```diff
@@ -478,6 +478,76 @@ let f (x : t) = x
     `);
   });
 
+  it("does not type-annotate already annotated argument", async () => {
```

Comment:
Could you try porting this test to OCaml in e2e-new?

The typescript suite is our legacy test suite. We haven't yet ported it to all to OCaml, but we'd like to stop adding new tests at least.

Reply:
Sure thing.
The funny thing is, that with the new test harness I get weird "destruct" action.
For argument x in function `let f (x : int) = 1` it wants to destruct it into:
`((0 as x) : int) | ((_ as x) : int`
It's unrelated to my PR and I've managed to reproduce this behavior on master branch:
<img width="361" alt="image" src="https://user-images.githubusercontent.com/22241847/222956097-e261d950-e937-4771-8d1f-a7a4ae7d17ce.png">

I'm not sure why there is a difference between results for e2e and e2e-new, but I guess it's something we should investigate.

Reply:
@rgrinberg , sure! I have ported them. After this PR gets merged I will port other type-annotate as well, so that they're all in one place. I already have the filtering functionality, so it should be a simple refactor.